### ISSUE TITLE
How External Auth works

### ISSUE THREAD

**Issue by github-germ** [MISP/MISP#5664]
I need help getting External Authentication working for auto login into MISP. I have an external user auth system that can pass the WebUI configured external auth key into MISP, but I am so far not able to get this to work as MISP keeps replying with the login form on `/users/login`. I don't see any `action=auth-fail` in the db `logs` table. Yes, I do have the `CustomAuth` settings correct best I can tell.

I read the following docs. If there's something more informative, please let me know:
* https://github.com/MISP/misp-book/blob/master/appendices/README.md which states something that does not seem to align with my testing so far, i.e.
* and also https://github.com/MISP/MISP/issues/3059

Questions:
1. What path should the `GET` with the `Authorization` HTTP header access?
2. What value should be contained in the `Authorization`? I thought it should be the same plain text value entered into that user's ext auth key in the WebUI; however,
   * From my quick review of the PHP code and based on what I see in the database `users.authkey` col seems to be a random 40-byte representation of the WebUI ext auth key is stored. 
   * And the README does not seem to talk about having to encode the key, i.e.
      > To create an external authenticated user, simply tick the External authentication user checkbox, after which an external auth key field will appear. This will be used to identify the users via the passed along header.

I also tested with `wget` passing in the HTTP header to various paths (`/users/login`, `/`, `index.php`) using either the plain text WebUI configured ext auth key for that user or what I see in the db `users.authkey` and still get the login form.

Lastly, I enabled Apache `log_forensic_module` and do see this `/var/www/MISP/app/webroot/.htaccess` line executing:
```
RewriteRule .* - [env=HTTP_AUTHORIZATION:%{HTTP:Authorization},last]
```
e.g.
```
[Sun Feb 23 16:14:58.248370 2020] [rewrite:trace5] [pid 15923] mod_rewrite.c(470): [client 192.168.230.201:30204] 192.168.230.201 - - [misp/sid#560e53fccfb8][rid#560e540d3138/initial/redir#1] setting env variable 'HTTP_AUTHORIZATION' to 'REDACTED'
```

**Comment by RichieB2B** [comment]
The Authorization header is only for API calls, not the web GUI.

**Comment by iglocska** [comment]
Replied via e-mail, but to recap what happened:

Authorization CAN be indeed used via the external auth, the check happens before the regular API key lookup - though it's recommended to use a different header.

What went wrong in this case was simply the display of the currently configured settings for the header namespace in server settings:

```
Plugin.CustomAuth_use_header_namespace
Plugin.CustomAuth_header_namespace
````

The above settings display a sane default (true and "HTTP_") which are ignored, so unless the settings are manually confirmed they default to false and "".

**Comment by github-germ** [comment]
We have external auth working now. Thanks for your help.

**Comment by vineethsai** [comment]
@github-germ Can you let us all know how you guys got it to work? Thank you!

**Comment by github-germ** [comment]
> @github-germ Can you let us all know how you guys got it to work? Thank you!

@vineethsai what is your external auth system? ours is home grown where we had to do additional settings in apache to redirect to our authenticator which came back with the user name which mapped to `users.external_auth_key` in the db. we set MISP `CustomAuth` settings as documented and had `external_auth_required` and `external_auth_key` set appropriately in the users db table.
